problems/4.arrays.js

```diff
@@ -1,31 +1,43 @@
 /** Averages the first and last value of an array.  Return null if the array has fewer than 2 elements.
-* @param {number[]} arr - The input array
-* @returns {number} - The average of the first and last values
-*/
-function firstAndLastAverage() {
-
+ * @param {number[]} arr - The input array
+ * @returns {number} - The average of the first and last values
+ */
+function firstAndLastAverage(arr) {
+  if (arr.length >= 2) {
+    return (arr.shift() + arr.pop()) / 2;
+  }
+  return null;
 }
 
 /** Duplicates all values inside an array.
-* Sample input: [1,4,2,5]
-* Sample output: [1,4,2,5,1,4,2,5]
-* @param {number[]} arr - The input array
-* @returns {number[]} - The duplicated array
-*/
-function duplicateArray() {
-
+ * Sample input: [1,4,2,5]
+ * Sample output: [1,4,2,5,1,4,2,5]
+ * @param {number[]} arr - The input array
+ * @returns {number[]} - The duplicated array
+ */
+function duplicateArray(arr) {
+  let newArray = [];
+  let arr2 = [...arr];
+  newArray = arr1.concat(arr2);
```

Comment:
I see an arr1 here but it's not declared anywhere else in the function. What array is this?